db/dcrpg/queries.go

```diff
@@ -356,15 +362,63 @@ func InsertVotes(db *sql.DB, dbTxns []*dbtypes.Tx, _ /*txDbIDs*/ []uint64, fTx *
 		return nil, nil, nil, nil, nil, err
 	}
 
+	// Prepare agenda votes insert statement.
+	agendaVotesInsert := internal.MakeAgendaVotesInsertStatement(checked)
+	agendaVotesStmt, err := dbtx.Prepare(agendaVotesInsert)
+	if err != nil {
+		log.Errorf("Agenda Votes INSERT prepare: %v", err)
+		_ = voteStmt.Close()
+		_ = dbtx.Rollback() // try, but we want the Prepare error back
+		return nil, nil, nil, nil, nil, err
+	}
+
 	bail := func() {
 		// Already up a creek. Just log any Rollback error.
 		_ = voteStmt.Close()
 		_ = agendaStmt.Close()
+		_ = agendaVotesStmt.Close()
 		if errRoll := dbtx.Rollback(); errRoll != nil {
 			log.Errorf("Rollback failed: %v", errRoll)
 		}
 	}
 
+	// Attempt to update the storedAgenda map is help keep track of the
+	// lockedIn, hardForked and activated heights.
+	if len(storedAgendas) == 0 {
+		var id int64
+		storedAgendas, err = retrieveAllAgendas(db)
+		if err != nil {
+			bail()
+			return nil, nil, nil, nil, nil,
+				fmt.Errorf("retrieveAllAgendas failed: : %v", err)
+		}
+
+		if storedAgendas == nil {
+			storedAgendas = make(map[string]dbtypes.MileStone)
+		}
+
+		for name, d := range votesMilestones.AgendaMileStones {
+			m, ok := storedAgendas[name]
+			if !ok || (d.Status != m.Status) {
```

Comment:
docs for the status check are helpful (e.g. `if there is a status difference between the milestone info from RPC and the cached milestone data,...`).  Just writing the comment makes me wonder how m.Status would be anything but the empty string since we only got to this part of the code because `storedAgendas == nil` and we just initialized the map.